File: app/src/main/java/io/github/getsmp/lorforandroid/ui/section/tracker/TrackerItemFactory.java
/*
 * Copyright 2016 getsmp
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.github.getsmp.lorforandroid.ui.section.tracker;

import android.text.Html;

import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.util.List;

import io.github.getsmp.lorforandroid.ui.section.Item;
import io.github.getsmp.lorforandroid.ui.section.ItemFactory;
import io.github.getsmp.lorforandroid.util.StringUtils;

public class TrackerItemFactory implements ItemFactory {
    @Override
    public void prepareItems(Element body, List items) {
        Elements topics = body.select("tbody > tr");
        for (Element topic : topics) {
            items.add(new Item(
                    topic.select("td:eq(1)").select("a").first().attr("href").substring(1),
                    Html.fromHtml(topic.select("td:eq(1)").select("a").first().ownText()).toString(),
                    topic.select("a.secondary").first().ownText(),
                    StringUtils.tagsFromElements(topic.select("span.tag")),
                    topic.select("time").first().ownText(),
                    topic.select("td.dateinterval > time").first().nextSibling().toString().replace(", ", ""),
                    StringUtils.numericStringToHumanReadable(topic.select("td.numbers").first().ownText())
            ));
        }
    }
}

File: app/src/main/java/io/github/getsmp/lorforandroid/ui/section/tracker/TrackerItemFactory.kt
/*
 * Copyright (c) 2016 getsmp
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package io.github.getsmp.lorforandroid.ui.section.tracker

import android.text.Html
import io.github.getsmp.lorforandroid.ui.section.Item
import io.github.getsmp.lorforandroid.ui.section.ItemFactory
import io.github.getsmp.lorforandroid.util.StringUtils
import org.jsoup.nodes.Element

class TrackerItemFactory : ItemFactory {
    override fun prepareItems(body: Element, items: MutableList<Any>) {
        val topics = body.select("tbody > tr")
        for (topic in topics) {
            items.add(Item(
                    url = topic.select("td:eq(1)").select("a").first().attr("href").substring(1),
                    title = Html.fromHtml(topic.select("td:eq(1)").select("a").first().ownText()).toString(),
                    groupTitle = topic.select("a.secondary").first().ownText(),
                    tags = StringUtils.tagsFromElements(topic.select("span.tag")),
                    date = topic.select("time").first().ownText(),
                    author = topic.select("td.dateinterval > time").first().nextSibling().toString().replace(", ", ""),
                    comments = StringUtils.numericStringToHumanReadable(topic.select("td.numbers").first().ownText())))
        }
    }
}
